### PR TITLE
Refactor presenters - Part 1

### PR DESCRIPTION
**Pull request by vipulnsward**
What:
- Refactor Presenters with small changes

Why:
- For consistency across them

For future:
- We intend to move to GraphQL instead once ReactJS migration is done

### CODE REVIEW

File: app/presenters/client/index_presenter.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-class ClientsPresenter
+class Client::IndexPresenter
```
Comment (vipulnsward) [inline app/presenters/client/index_presenter.rb:3]: @shalapatil for now, I grouped them under clients folder.

File: app/presenters/client/index_presenter.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ def initialize(clients, current_company, current_user)
     @current_company = current_company
   end
 
-  def group_projects_by_client_name
+  def projects_grouped_by_client_name
```
Comment (vipulnsward) [inline app/presenters/client/index_presenter.rb:12]: These are for better readability

File: app/presenters/client/show_presenter.rb [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-class ClientPresenter
+class Client::ShowPresenter
   attr_reader :client
 
   def initialize(client)
     @client = client
   end
 
-  def snippet
+  def process
```
Comment (vipulnsward) [inline app/presenters/client/show_presenter.rb:10]: We can expand on the data sent out from here for individual resources as needed later.

File: app/presenters/project_members_presenter.rb [hunk 4]
```diff
@@ -8,14 +8,14 @@ def initialize(projects)
   end
 
   def project_to_member_hourly_rate
-    projects_members_rates ||= projects.group_by(&:id).transform_values do |project|
+    @_project_to_member_hourly_rate ||= projects.group_by(&:id).transform_values do |project|
```
Comment (vipulnsward) [inline app/presenters/project_members_presenter.rb:11]: Fixes memoization. Secondly the variable name should match the method name

**Comment by vipulnsward** [comment]
@shalapatil please review, add specs and merge.

File: app/presenters/client/index_presenter.rb (hunk 1 above)
Comment (shalapatil) [reply app/presenters/client/index_presenter.rb:3]: 👍

Comment (shalapatil) [reply app/presenters/client/index_presenter.rb:3]: 👍

File: app/presenters/client/index_presenter.rb (hunk 2 above)
Comment (shalapatil) [reply app/presenters/client/index_presenter.rb:12]: if you are naming process in other presenter then why to use some other name here?

File: app/presenters/project_members_presenter.rb (hunk 4 above)
Comment (shalapatil) [reply app/presenters/project_members_presenter.rb:11]: yeah, my bad.

**Pull request merged**